testkit/appkit/app.go

```diff
@@ -93,41 +92,75 @@ func (ak *AppKit) GetNodeId(home string) (string, error) {
 	return ak.execCmd([]string{"tendermint", "show-node-id", "--home", home})
 }
 
-func (ak *AppKit) StartNode(home string) error {
+func (ak *AppKit) StartNode(home, loglvl string) error {
 	ak.Cmd.ResetFlags()
 
 	ak.Cmd.SetErr(os.Stdout)
-	ak.Cmd.SetArgs([]string{"start", "--home", home, "--log_level", "info"})
+	ak.Cmd.SetArgs([]string{"start", "--home", home, "--log_level", loglvl, "--log_format", "plain"})
 
 	return svrcmd.Execute(ak.Cmd, appcmd.EnvPrefix, app.DefaultNodeHome)
 }
 
-func GetBlockHashByHeight(ip net.IP, height int) (string, error) {
-	uri := fmt.Sprintf("http://%s:26657/block?height=%d", ip.To4().String(), height)
+func (ak *AppKit) PayForData(accAdr string, msg int, krbackend, chainId, home string) error {
+	ak.Cmd.ResetFlags()
+	ak.Cmd.SetArgs([]string{
+		"tx", "payment", "payForData", fmt.Sprint(msg),
+		"--from", accAdr, "-b", "block", "-y", "--gas", "1000000000",
+		"--fees", "100000000000utia", //15
+		"--keyring-backend", krbackend, "--chain-id", chainId, "--home", home, "--keyring-dir", home,
+	})
+
+	return svrcmd.Execute(ak.Cmd, appcmd.EnvPrefix, app.DefaultNodeHome)
+}
+
+func getResultBlockResponse(uri string) (coretypes.ResultBlock, error) {
 	resp, err := http.Get(uri)
 	if err != nil {
-		return "", err
+		return coretypes.ResultBlock{}, err
 	}
 	defer resp.Body.Close()
+
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return "", err
+		return coretypes.ResultBlock{}, err
 	}
 
 	var rpcResponse types.RPCResponse
 	if err := rpcResponse.UnmarshalJSON(body); err != nil {
-		return "", err
+		return coretypes.ResultBlock{}, err
 	}
 
 	var resBlock coretypes.ResultBlock
 	if err := tmjson.Unmarshal(rpcResponse.Result, &resBlock); err != nil {
+		return coretypes.ResultBlock{}, err
+	}
+
+	return resBlock, nil
+}
+
+func GetBlockHashByHeight(ip net.IP, height int) (string, error) {
+	uri := fmt.Sprintf("http://%s:26657/block?height=%d", ip.To4().String(), height)
+
+	resBlock, err := getResultBlockResponse(uri)
+	if err != nil {
 		return "", err
 	}
 
 	return resBlock.BlockID.Hash.String(), nil
 }
 
-func updateConfig(path, key, value string) error {
+func GetLatestsBlockSize(ip net.IP) (int, error) {
+	uri := fmt.Sprintf("http://%s:26657/block", ip.To4().String())
+
+	resBlock, err := getResultBlockResponse(uri)
+	if err != nil {
+		return 0, err
+	}
+
+	return resBlock.Block.Size(), nil
+}
+
+func updateConfig(path, key string, value interface{}) error {
```

Comment:
what is the reason of changing `string` to `interface{}`? As far as I see, you are passing strings to updateConfig everywhere in code

Reply:
Yes. The main reason lies in the config.toml itself. There are fields where there are bool and others